### PR TITLE
[CLOUDGA-18071] CLI support for asymmetric geo-partitioned cluster creation

### PR DESCRIPTION
**Pull request by vmallepalli**
### Summary
- Added support for asymmetric geo-partitioned cluster creation (different regions can now have different num-cores, disk-size-gb, disk-iops for different regions - only for geo-partitioned clusters)
- New flows are hidden behind the ASYMMETRIC_GEO feature flag

### Testing
- Successfully created, updated, and described asymmetric geo cluster
- Correctly fails when attempting to create asymmetric synchronous cluster

The new payloads look like:

```
./ybm cluster create --cluster-name asymmetric-geo-cli-2 --credentials username=vm,password=pwd --cloud-provider AWS --cluster-tier Dedicated --fault-tolerance REGION --region-info region=us-west-2,num-nodes=1,vpc=aws-west-2,num-cores=2 --region-info region=us-east-2,num-nodes=1,vpc=aws-east-2,num-cores=4 --region-info region=ca-central-1,num-nodes=1,vpc=aws-central-1,num-cores=8,disk-size-gb=500,disk-iops=4000 --cluster-type GEO_PARTITIONED
```

```
./ybm cluster update --cluster-name asymmetric-geo-cli-1 --region-info region=us-west-2,num-nodes=2,vpc=aws-west-2,num-cores=4,disk-iops=4000 --region-info region=us-east-2,num-nodes=2,vpc=aws-east-2,num-cores=4,disk-size-gb=700 --region-info region=ca-central-1,num-nodes=2,vpc=aws-central-1,num-cores=8,disk-size-gb=600,disk-iops=3000

```

### CODE REVIEW

**Review by posriniv — COMMENTED** [review]
Would wait for a review from @shinji62 or @cdavid too

File: cmd/cluster/create_cluster.go
Comment (vmallepalli) [reply cmd/cluster/create_cluster.go:59]: The node-config flag is marked deprecated, so whenever it's used a message pops up to instead use region-info. 
Also, the node-config flag no longer appears in help or usage messages, so new users will never use this flag. 
For existing users who may not had a chance to move over yet, imo we should allow them to continue using node-config and give them time to switch over. 
Behavior here is to use flags from region-info when provided, and fallback to the older deprecated node-config flag.

**Review by posriniv — APPROVED** [review]
Looks good to me!

**Comment by shinji62** [comment]
@vmallepalli is test are ok on your side ?

File: go.mod
Comment (kc-yb) [inline go.mod:24]: Please fix this

File: go.sum
Comment (kc-yb) [inline go.sum:285]: merge errors

**Comment by kc-yb** [comment]
- Can you add an example of the CLI on how the command looks like to configure an asymmetric cluster just to get an idea of how complex is it to configure from CLI?
- With the number of options increasing, CLI options and configurations are getting more complex for cluster creation (esp with nested arguments), will it make sense to take an input from a config file for cluster creation (future enhancement)? @cdavid

**Comment by posriniv** [comment]
Looks okay to me. Let's wait for another approval form either @kc-yb or @shinji62 .

**Review by kc-yb — APPROVED** [review]
LG overall, please wait for approval from @shinji62

**Comment by shinji62** [comment]
Approved

**Review by shinji62 — APPROVED** [review]

**Pull request merged**
